x/foundation/module/module.go

```diff
@@ -0,0 +1,258 @@
+		panic(err)
+	}
+	return []abci.ValidatorUpdate{}
+}
+
+// ExportGenesis returns the exported genesis state as raw bytes for the foundation
+// module.
+func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage {
+	gs := am.keeper.ExportGenesis(ctx)
+	return cdc.MustMarshalJSON(gs)
+}
+
+// ConsensusVersion implements AppModule/ConsensusVersion.
+func (AppModule) ConsensusVersion() uint64 { return consensusVersion }
+
+// BeginBlock performs a no-op.
+func (am AppModule) BeginBlock(ctx context.Context) error {
+	return keeper.BeginBlocker(sdk.UnwrapSDKContext(ctx), am.keeper)
+}
+
+// EndBlock performs a no-op.
+func (am AppModule) EndBlock(ctx context.Context) error {
+	return keeper.EndBlocker(sdk.UnwrapSDKContext(ctx), am.keeper)
+}
+
+// ____________________________________________________________________________
+
+// AppModuleSimulation functions
+
+// // GenerateGenesisState creates a randomized GenState of the foundation module.
+// func (AppModule) GenerateGenesisState(simState *module.SimulationState) {
+// 	simulation.RandomizedGenState(simState)
+// }
+
+// // ProposalContents returns all the foundation content functions used to
+// // simulate foundation proposals.
+// func (AppModule) ProposalContents(simState module.SimulationState) []simtypes.WeightedProposalContent {
+// 	return simulation.ProposalContents()
+// }
+
+// // RandomizedParams creates randomized foundation param changes for the simulator.
+// func (AppModule) RandomizedParams(r *rand.Rand) []simtypes.ParamChange {
+// 	return simulation.ParamChanges(r)
+// }
+
+// // RegisterStoreDecoder registers a decoder for foundation module's types
+// func (am AppModule) RegisterStoreDecoder(sdr sdk.StoreDecoderRegistry) {
+// 	sdr[types.StoreKey] = simulation.NewDecodeStore(am.cdc)
+// }
+
+// // WeightedOperations returns the all the foundation module operations with their respective weights.
+// func (am AppModule) WeightedOperations(simState module.SimulationState) []simtypes.WeightedOperation {
+// 	return simulation.WeightedOperations(
+// 		simState.AppParams, simState.Cdc,
+// 		am.stakingKeeper, am.slashingKeeper, am.keeper, simState.Contents,
+// 	)
+// }
+
+// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
+func (am AppModule) IsOnePerModuleType() {}
+
+// IsAppModule implements the appmodule.AppModule interface.
+func (am AppModule) IsAppModule() {}
+
+func init() {
+	appmodule.Register(&modulev1.Module{},
+		appmodule.Provide(ProvideModule, ProvideKeyTable),
+	)
+}
+
+type FoundationInputs struct {
+	depinject.In
+
+	StoreService store.KVStoreService
+	Cdc          codec.Codec
+	Config       *modulev1.Module
+
+	AuthKeeper       foundation.AuthKeeper
+	BankKeeper       foundation.BankKeeper
+	Subspace         paramstypes.Subspace
+	MsgServiceRouter baseapp.MessageRouter
+}
+
+type FoundationOutputs struct {
+	depinject.Out
+
+	Keeper keeper.Keeper
```

Comment:
Please set a foundation keeper as struct. Stakingplus module can't use foundation keeper because foundation keeper struct is not register to simapp. Some tests of simapp has following error.

```
--- FAIL: TestInitCmd (0.15s)
panic: can't resolve type github.com/Finschia/finschia-sdk/x/stakingplus/stakingplus.FoundationKeeper for github.com/Finschia/finschia-sdk/x/stakingplus/module.ProvideModule
```

Reply:
I've updated the keeper.